.travis.yml

```diff
@@ -28,122 +23,131 @@ jobs:
+    - stage: build
       os: linux
+      name: Linux AMD64 Unit Test
+      arch: amd64
+      if: type = pull_request
       script:
-        - scripts/travis/build_test.sh
-
-    - stage: build_pr
+        - scripts/travis/travis_wait.sh 90 mule build-linux-amd64-shorttest
+    - stage: build
       os: linux
-      name: Ubuntu AMD64 Build
+      name: Linux AMD64 Unit Test
+      arch: amd64
+      if: type != pull_request
       script:
-        - scripts/travis/build_test.sh
+        - scripts/travis/travis_wait.sh 120 mule build-linux-amd64-fulltest
     - # same stage, parallel job
       os: linux
-      name: Ubuntu AMD64 Integration Test
+      name: Linux AMD64 Integration Test
+      arch: amd64
       script:
-        - ./scripts/travis/integration_test.sh
+        - mule build-linux-amd64
+        - scripts/travis/travis_wait.sh 120 mule build-linux-amd64-integration
     - # same stage, parallel job
-      name: External ARM64 Build
+      name: Linux ARM64 Unit Test
       os: linux
-      env:
-        - BUILD_TYPE: "external_build"
-        - TARGET_PLATFORM: "linux-arm64"
-      addons:
-        apt:
-          packages:
-            - awscli
+      arch: arm64
+      if: type = pull_request
       script:
-        - scripts/travis/external_build.sh ./scripts/travis/build_test.sh
+        - scripts/travis/travis_wait.sh 90 mule build-linux-arm64-shorttest
     - # same stage, parallel job
-      name: External ARM64 Integration Test
+      name: Linux ARM64 Unit Test
       os: linux
-      env:
-        - BUILD_TYPE: "external_build"
-        - TARGET_PLATFORM: "linux-arm64"
-      addons:
-        apt:
-          packages:
-            - awscli
+      arch: arm64
+      if: type != pull_request
       script:
-        - scripts/travis/external_build.sh ./scripts/travis/integration_test.sh
+        - scripts/travis/travis_wait.sh 120 mule build-linux-arm64-fulltest
     - # same stage, parallel job
-      os: osx
-      osx_image: xcode11
-      name: MacOS AMD64 Build
+      name: Linux ARM64 Integration Test
+      os: linux
+      arch: arm64
       script:
-        - scripts/travis/build_test.sh
+        - mule build-linux-arm64
+        - scripts/travis/travis_wait.sh 120 mule build-linux-arm64-integration
     - # same stage, parallel job
       os: osx
       osx_image: xcode11
-      name: MacOS AMD64 Integration Test
-      script:
-        - ./scripts/travis/integration_test.sh
-
-    - stage: build_release
-      os: linux
-      name: Ubuntu AMD64 Build
-      script:
-        - ./scripts/travis/build_test.sh
-    - # same stage, parallel job
-      os: linux
-      name: Ubuntu AMD64 Integration Test
+      arch: amd64
+      name: MacOS AMD64 Unit Test
+      go:
+        - "1.12"
+      go_import_path: github.com/algorand/go-algorand
+      language: go
+      before_install:
+        - brew update
+        - brew upgrade pyenv
+        - pyenv install 3.7.6
+        - pyenv local  3.7.6
```

Comment:
Why do you need the go stuff, Isn't everything happening in a container now?

Reply:
We're having to execute the OSX builds locally since there aren't darwin containers unfortunately